bpf/process/types/basic.h

```diff
@@ -567,15 +566,11 @@ __filter_file_buf(char *value, char *args, __u32 op)
 		if (a < v)
 			goto skip_string;
 	} else if (op == op_filter_str_postfix) {
-#ifdef __LARGE_BPF_PROG
-		err = cmpbytes(&value[4], &args[4], v - 1);
-#else
-		err = cmpbytes_small(&value[4], &args[4], v - 1);
-#endif
+		err = rcmpbytes(&value[4], &args[4], v - 1, a - 1);
```

Comment:
why you use `rcmpbytes` here? See the comments for `filter_file_buf` 
https://github.com/cilium/tetragon/blob/e59e5d3e4c6156b2da26447cbb476a3d54ce559b/bpf/process/types/basic.h#L584

Seems that we should use `cmpbytes` when operation is `op_filter_str_postfix`, while the `rcmpbytes` compare strings from the right side.

But I'm not sure if this matters... Since you have verified the postfix in the following tests.

Reply:
Yes, the comment is outdated. Before https://github.com/cilium/tetragon/pull/90 the paths are stored in reverse order as the comment says. https://github.com/cilium/tetragon/pull/90 stores the paths in the correct order and now we have to do a reverse search only for postfix. We can handle Equal and Prefix with `cmpbytes`.

I will cleanup the comment.